pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -1612,12 +1613,63 @@ func TestApplyWorkspaces(t *testing.T) {
 
 func TestApplyTaskResultsToPipelineResults(t *testing.T) {
 	for _, tc := range []struct {
-		description string
-		results     []v1beta1.PipelineResult
-		taskResults map[string][]v1beta1.TaskRunResult
-		runResults  map[string][]v1alpha1.RunResult
-		expected    []v1beta1.PipelineRunResult
+		description     string
+		results         []v1beta1.PipelineResult
+		taskResults     map[string][]v1beta1.TaskRunResult
+		runResults      map[string][]v1alpha1.RunResult
+		expectedResults []v1beta1.PipelineRunResult
+		expectedError   error
 	}{{
+		description: "non-reference-results",
+		results: []v1beta1.PipelineResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewArrayOrString("resultName"),
+		}},
+		taskResults: map[string][]v1beta1.TaskRunResult{
+			"pt1": {
+				{
+					Name:  "foo",
+					Value: *v1beta1.NewArrayOrString("do", "rae", "mi"),
+				},
+			},
+		},
+		expectedResults: nil,
+	}, {
+		description: "array-index-out-of-bound",
```

Comment:
it looks like there are some test cases for arrays being added. Could you please update the PR to just focus on object pipeline results, and make any changes to arrays separately?

Reply:
removed the array validation part